### PR TITLE
Use `rfc2822` date format; resolves #99

### LINKED ISSUE #99: Use `rfc2822` date format
Changing `git-fast-import(1)` `--date-format` from `raw` to `rfc22822` to support more `GIT_COMMITTER_DATE` date values.

PR: #100

----

Rationale:

In 2faa33c (Add support for `GIT_COMMITTER_(NAME|EMAIL|DATE)` env vars; resolves 98, 2022-04-27) the `GIT_COMMITTER_DATE` environment parameter was introduced.

The `GIT_COMMITTER_DATE` environment parameter [ref](https://git-scm.com/docs/git-commit#_date_formats) supports three formats:

1. Git internal format [ref](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-Gitinternalformat)
2. RFC 2822 [ref](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-RFC2822)
3. ISO 8601 [ref](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-ISO8601)

The `git-fast-import(1)` [ref](https://git-scm.com/docs/git-fast-import) comamnds `--date-format` option [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt---date-formatltfmtgt) supports four _Date formats_ [ref](https://git-scm.com/docs/git-fast-import#_date_formats) of which two are interesting:

1. Git internal format `raw` [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt-coderawcode)
2. RFC 2822 `rfc2822` [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt-coderfc2822code)

Currently in 2faa33c `--date-format=raw` is in use:

> Unlike the `rfc2822` format, this format is very strict. Any variation in formatting will cause fast-import to reject the value, and some sanity checks on the numeric values may also be performed.

This renders two out of three supported formats in `GIT_COMMITTER_DATE` as unsupported: RFC 2822 and ISO 8601, only the very strict _Git internal format_ works.

On the other hand, when using `git-fast-import(1)` with `--date-format=rfc2822`, three out of three supported formats in `GIT_COMMITTER_DATE` are supported (100%), as it accepts both _Git internal format_ as well as _ISO 8601_ as date formats.

Additionally in build context, `raw` does not support building at the beginning of the epoch (`0 +0000`), as with the Git internal format it must be after Jan 1, 2000 (git v0.99) / Mar 3 1973 (git v1.5.2.2).  With `rfc2822` it is possible to express this intend (`Thu, 01 Jan 1970 00:00:00 +0000`, `1970-01-01 00:00:00 +0000` etc.).

---

Notes:

1. Tested against git v2.25.1
2. _Date format_ `raw-permissive` was not available with my version ("fatal: unknown --date-format argument raw-permissive") [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt-coderaw-permissivecode)
3. _Date format_ `now` does not allow to specify the _Git Comitter Date_ (_"This is a toy format."_) [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt-codenowcode)
4. This improvement ~~is to be~~ suggested upstream to ghp-import: https://github.com/c-w/ghp-import/issues/108
5. `git_utils.make_when`s `time.time()` reflecting `SOURCE_DATE_EPOCH` deferred

### PR DESCRIPTION
**Pull request by ktomk**

### CODE REVIEW

File: mike/git_utils.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def copy(self, destdir='', start=''):
 
 class Commit:
     def __init__(self, branch, message):
-        cmd = ['git', 'fast-import', '--date-format=raw', '--quiet', '--done']
+        cmd = ['git', 'fast-import', '--date-format=rfc2822', '--quiet', '--done']
```
Comment (jimporter) [inline mike/git_utils.py:158]:
```suggestion
        cmd = ['git', 'fast-import', '--date-format=rfc2822', '--quiet',
               '--done']
```

**Review by jimporter — CHANGES_REQUESTED** [review]
This seems like a good change, thanks. It could probably use some tests, but I should have added those when I added support for `GIT_COMMITTER_DATE`, so I'll try and get to that in the next few days.

Just one small change to fix the lint failure...

**Comment by ktomk** [comment]
Update w/ lint fix and on top of master.

Regarding tests, "upstream" ghp-import took in the change already and [at the end of the issue description there I left some notes on how I tested it](https://github.com/c-w/ghp-import/issues/108) if it helps.

I find this hard to test on the level of unit-tests b/c of the subprocess, IMHO this is more an integration check with the git utility and the environment. git/git-fast-import should work as expected. Perhaps a suite that supports setting the environment and binaries w/ a git repo fixture to poke around with.

**Comment by jimporter** [comment]
Thanks, merged!

**Pull request merged**